packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -161,7 +157,7 @@ class MockConnection implements Connection {
 
   waitForWriteRequest(): Promise<api.WriteRequest> {
     if (this.earlyWrites.length > 0) {
-      return Promise.resolve(this.earlyWrites.shift()) as AnyDuringMigration;
+      return Promise.resolve(this.earlyWrites.shift()!);
```

Comment:
To avoid `!` you could probably rewrite this as:

```
const earlyWrite = this.earlyWrites.shift();
if (earlyWrite) {
  return Promise.resolve(earlyWrite);
}
```

Up to you though. I like to avoid `!`.

Reply:
Oh yeah, good suggestion. I like that better.